src/devices/nm-device.c

```diff
@@ -2181,6 +2181,34 @@ _get_mdns (NMDevice *self)
 	return mdns;
 }
 
+static NMSettingConnectionLlmnr
+_get_llmnr (NMDevice *self)
+{
+	NMConnection *connection;
+	NMSettingConnectionLlmnr llmnr = NM_SETTING_CONNECTION_LLMNR_DEFAULT;
+
+	g_return_val_if_fail (NM_IS_DEVICE (self), NM_SETTING_CONNECTION_LLMNR_DEFAULT);
+
+	connection = nm_device_get_applied_connection (self);
+	if (connection)
+		llmnr = nm_setting_connection_get_llmnr (nm_connection_get_setting_connection (connection));
+
+	if (llmnr == NM_SETTING_CONNECTION_LLMNR_DEFAULT) {
+		gs_free char *value = NULL;
+
+		value = nm_config_data_get_connection_default (NM_CONFIG_GET_DATA,
+		                                               "connection.llmnr",
+		                                               self);
+		llmnr = _nm_utils_ascii_str_to_int64 (value,
```

Comment:
maybe it's time to add a helper `nm_config_data_get_connection_default_int64()`? We are doing this repeatedly.

Reply:
How about now?